Celeste.Mod.mm/Mod/Helpers/TypeHelper.cs

```diff
@@ -213,37 +208,85 @@ public static class TypeHelper {
                 if(overwrite || !entityDataNameToType.ContainsKey(kvp.Key))
                     entityDataNameToType[kvp.Key] = kvp.Value;
             }
         }
-
-        public static bool GetTypeFromDataName(string entityDataName, out Type type) {
-            return entityDataNameToType.TryGetValue(entityDataName, out type);
-        }
-
+        /// <summary>
+        /// Given an Entity, checks whether the Name of the EntityData that constructed it (if any exists) matches the string provided
+        /// </summary>
+        /// <param name="entity">an Entity</param>
+        /// <param name="entityDataName">an EntityData Name to compare to</param>
+        /// <returns>whether the EntityData that constructed the Entity (if it exists) matches the provided string</returns>
         public static bool CheckEntityByDataName(Entity entity, string entityDataName) {
-            return (entity as patch_Entity).EntityData is { } ed && ed.Name == entityDataName;
+            return (entity as patch_Entity).EntityData?.Name == entityDataName;
         }
-
+        /// <summary>
+        /// Checks whether an Entity has an EntityData and if that EntityData matches a criterion. 
+        /// </summary>
+        /// <param name="entity">an Entity</param>
+        /// <param name="predicate">what the EntityData within that Entity "should" meet the criterion of</param>
+        /// <returns>whether the Entity meets that EntityData criterion</returns>
         public static bool CheckEntityByData(Entity entity, Predicate<EntityData> predicate) {
             return (entity as patch_Entity).EntityData is { } ed && predicate(ed);
         }
-        public static Type GetType(string fullname, bool cache = true) {
-            if(typeFullNameToTypeCache.ContainsKey(fullname)) { return typeFullNameToTypeCache[fullname]; }
-            Type type = FakeAssembly.GetFakeEntryAssembly().GetType(fullname);
-            if(type != null) {
-                typeFullNameToTypeCache[fullname] = type;
-                return type;
+
+        /// <summary>
+        /// "Safe" variant of GetType. Attempts to get the type (class) based on the *full name* of the Type provided. Use this in your `if` statements, instead of GetType
+        /// </summary>
+        /// <param name="fullname">The Full Name (Name including namespaces) of the Type you are trying to find. Example: Celeste's Level class is 'Celeste.Level'</param>
+        /// <param name="type">The type as a class found. Returns null if not found</param>
+        /// <param name="cache">Set this to false if you have no intention of obtaining this type with this method after your first use.</param>
+        /// <returns>Whether or not a type (class) was able to be found given the provided FullName</returns>
+        public static bool TryGetType(string fullname, out Type type, bool cache = true) {
+            if (FullName_to_Type.ContainsKey(fullname)) {
+                type = FullName_to_Type[fullname];
```

Comment:
`TryGetValue` to avoid doing 2 dictionary lookups here

Reply:
At first I thought TryGetValue and ContainsKey used two different lookup methods but that's apparently either just wrong or some .NET FW jank so yeah I'll update all of these scenarios